index.html

```diff
@@ -1,14 +1,77 @@
 <!DOCTYPE html>
 <html lang="en">
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Document</title>
-    
-    <link rel="stylesheet" href="css/styles.css">
-</head>
-<body>
-    
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Currency Converter</title>
+    <link rel="stylesheet" href="css/styles.css" />
+  </head>
+  <body>
+    <div class="container">
+      <h1>Currency Converter</h1>
+      <!-- Form for inserting a new currency rate -->
+      <div class="form-container">
+        <form id="newRateForm">
+          <h2>Insert New Currency Rate</h2>
+          <label for="baseCurrency">Base Currency:</label>
+          <input type="text" id="baseCurrency" name="baseCurrency" required />
+          <label for="targetCurrency">Target Currency:</label>
+          <input
+            type="text"
+            id="targetCurrency"
+            name="targetCurrency"
+            required
+          />
+          <label for="rate">Rate:</label>
+          <input type="number" step="0.1" id="rate" name="rate" required />
+          <button type="button" onclick="insertRate()">Insert Rate</button>
```

Comment:
Multiple things here:
- The role of this button is to submit its parent form. So define it as a `type="submit"`. 
- Prefer attaching events to a DOM element using `addEventListener` instead of HTML attributes
- When dealing with forms, prefer using the [submit event of the form](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event) instead of a click on a button.

This comment is, of course, valid for the other forms in this file.